ui/app/AppLayouts/Communities/panels/SharedAddressesAccountSelector.qml

```diff
@@ -161,25 +166,11 @@ StatusListView {
                 ButtonGroup.group: d.addressesGroup
                 anchors.verticalCenter: parent.verticalCenter
                 checkable: true
-                checked: root.selectedSharedAddresses.some((address) => address.toLowerCase() === listItem.address )
-                enabled: !(root.selectedSharedAddresses.length === 1 && checked) // last cannot be unchecked
-                onToggled: {
-                    // handle selected addresses
-                    const index = root.selectedSharedAddresses.findIndex((address) => address.toLowerCase() === listItem.address)
-                    const selectedSharedAddressesCopy = Object.assign([], root.selectedSharedAddresses) // deep copy
-                    if (index === -1) {
-                        selectedSharedAddressesCopy.push(listItem.address)
-                    } else {
-                        selectedSharedAddressesCopy.splice(index, 1)
-                    }
-                    root.selectedSharedAddresses = selectedSharedAddressesCopy
+                checked: root.selectedSharedAddressesMap.has(listItem.address)
+                enabled: !(d.selectedSharedAddressesCount === 1 && checked) // last cannot be unchecked
 
-                    // switch to next available airdrop address when unchecking
-                    if (!checked && listItem.address === root.selectedAirdropAddress.toLowerCase()) {
-                        d.selectFirstAvailableAirdropAddress()
-                    }
-
-                    root.addressesChanged()
+                onToggled: {
+                    root.toggleAddressSelection(model.keyUid, listItem.address)
```

Comment:
Wondering if it's not enough to share the `keyUid`? I guess it's the identifier of the accounts item and this item will also contain the `address` role??

Reply:
No it's not, cause there can be many addresses for a single key pair, but just a few of them are revealed.